core/src/test/java/hivemall/evaluation/FMeasureUDAFTest.java

```diff
@@ -0,0 +1,355 @@
+        boolean actual = true;
+        boolean predicted = false;
+        double beta = 1.0d;
+        binarySetUp(actual, predicted, beta, "binary");
+
+        evaluator.init(GenericUDAFEvaluator.Mode.PARTIAL1, inputOIs);
+        evaluator.reset(agg);
+
+        evaluator.iterate(agg, new Object[] {actual, predicted});
+
+        Assert.assertEquals(-1.d, agg.get(), 1e-4);
+    }
+
+    @Test(expected = HiveException.class)
+    public void testBinaryInvalidStringInput() throws Exception {
+        String actual = "cat";
+        int predicted = 1;
+        binarySetUp(actual, predicted, 1.0, "micro");
+
+        evaluator.init(GenericUDAFEvaluator.Mode.PARTIAL1, inputOIs);
+        evaluator.reset(agg);
+
+        evaluator.iterate(agg, new Object[] {actual, predicted});
+
+        agg.get();
+    }
+
+
+    @Test(expected = HiveException.class)
+    public void testBinaryInvalidLargeIntInput() throws Exception {
+        int actual = 1;
+        int predicted = 3;
+        binarySetUp(actual, predicted, 1.0, "micro");
+
+        evaluator.init(GenericUDAFEvaluator.Mode.PARTIAL1, inputOIs);
+        evaluator.reset(agg);
+
+        evaluator.iterate(agg, new Object[] {actual, predicted});
+
+        agg.get();
+    }
+
+
+    @Test(expected = HiveException.class)
+    public void testMultiLabelZeroBeta() throws Exception {
+        List<Integer> actual = Arrays.asList(1, 3, 2, 6);
+        List<Integer> predicted = Arrays.asList(1, 2, 4);
+        double beta = 0.;
+        setUpWithArguments(beta, "micro");
+
+        evaluator.init(GenericUDAFEvaluator.Mode.PARTIAL1, inputOIs);
+        evaluator.reset(agg);
+
+        evaluator.iterate(agg, new Object[] {actual, predicted});
+
+        // FMeasure for beta has zero value is not defined
+        agg.get();
+    }
+
+    @Test(expected = HiveException.class)
+    public void testMultiLabelNegativeBeta() throws Exception {
+        List<Integer> actual = Arrays.asList(1, 3, 2, 6);
+        List<Integer> predicted = Arrays.asList(1, 2, 4);
+        double beta = -1.0d;
+        String average = "micro";
+        setUpWithArguments(beta, average);
+
+        evaluator.init(GenericUDAFEvaluator.Mode.PARTIAL1, inputOIs);
+        evaluator.reset(agg);
+
+        evaluator.iterate(agg, new Object[] {actual, predicted});
+
+        // FMeasure for beta has negative value is not defined
+        agg.get();
+    }
+
+    @Test
+    public void testMultiLabelF1score() throws Exception {
+        List<Integer> actual = Arrays.asList(1, 3, 2, 6);
+        List<Integer> predicted = Arrays.asList(1, 2, 4);
+        double beta = 1.0;
+        String average = " micro";
+        setUpWithArguments(beta, average);
+
+        evaluator.init(GenericUDAFEvaluator.Mode.PARTIAL1, inputOIs);
+        evaluator.reset(agg);
+
+        evaluator.iterate(agg, new Object[] {actual, predicted});
+
+        Assert.assertEquals(0.5714285714285715d, agg.get(), 1e-5);
```

Comment:
Could you write how did you get the expected result? (Spark similarly to `testMultiLabelF1MultiSamples`?)